backend/build-and-copy.sh

```
@@ -24,24 +24,23 @@ else
   echo "❌ frontend directory not found in project root"
   echo "🔍 Available directories:"
-  ls -la
+  ls -la "$PROJECT_ROOT"
   exit 1
 fi
 
 echo "✅ frontend directory found"
 
 # frontendディレクトリに移動
-cd frontend
+cd "$PROJECT_ROOT/frontend"
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)